<img src="images/usm.png" width="480" height="240" align="left"/>

# MAT281 - 2° Semestre 2019
## Aplicaciones de la Matemática en la Ingeniería

EL siguiente laboratorio debe realizarse de manera individual. No es posible copiar y pegar código de otros estudiantes. 
Pueden consultar internet (stackoverflow, por ejemplo), pero deben indicar sus fuentes cuando corresponda.

**Nombre**: *Joaquín Silva*

**Rol**: *201604611-9*

## Objetivos del laboratorio

* Reforzar conceptos básicos de la librería pandas.


## Contenidos

* [Problema 01](#p1)



<a id='p1'></a>
## I.- Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información tal como: edad ,sexo, profesión, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [1]:
import pandas as pd
import os

# cargar datos
df = pd.read_csv(os.path.join("data","ocupation.csv"), sep="|")
df.head()

user_id  age gender  occupation zip_code
0        1   24      M  technician    85711
1        2   53      F       other    94043
2        3   23      M      writer    32067
3        4   24      M  technician    43537
4        5   33      F       other    15213

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:
1. ¿Cuál es el número de observaciones en el conjunto de datos?
2. ¿Cuál es el número de columnas en el conjunto de datos?
3. Imprime el nombre de todas las columnas
4. ¿Cómo se indexa el conjunto de datos?
5. ¿Cuál es el tipo de datos de cada columna?
6. Resumir el conjunto de datos (**Hint**: pd.describe())
7. Resume todas las columnas (**Hint**: pd.describe(all))
8. Imprimir solo la columna de ocupación
9. ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?
10. ¿Cuál es la ocupación más frecuente?
11. ¿Cuál es la edad media de los usuarios?
12. ¿Cuál es la edad con menos ocurrencia?

In [2]:
# Pregunta 1
df.shape

(943, 5)

Como hay 943 filas en el data frame, hay la misma cantidad de observaciones en el conjunto de datos.

In [3]:
# Pregunta 2
# El número de columnas es 5

# Pregunta 3
col = list(df.columns)
print(col)

['user_id', 'age', 'gender', 'occupation', 'zip_code']


In [5]:
# Pregunta 4
# El conjunto de datos está indexado por la columna 'user_id'.

# Pregunta 5
df.dtypes

user_id        int64
age            int64
gender        object
occupation    object
zip_code      object
dtype: object

In [6]:
# Las primeras 2 columnas son tipo int y las siguientes son tipo object

# Pregunta 6
df.describe()

user_id         age
count  943.000000  943.000000
mean   472.000000   34.051962
std    272.364951   12.192740
min      1.000000    7.000000
25%    236.500000   25.000000
50%    472.000000   31.000000
75%    707.500000   43.000000
max    943.000000   73.000000

In [7]:
# Pregunta 7
df.describe(include='all')

user_id         age gender occupation zip_code
count   943.000000  943.000000    943        943      943
unique         NaN         NaN      2         21      795
top            NaN         NaN      M    student    55414
freq           NaN         NaN    670        196        9
mean    472.000000   34.051962    NaN        NaN      NaN
std     272.364951   12.192740    NaN        NaN      NaN
min       1.000000    7.000000    NaN        NaN      NaN
25%     236.500000   25.000000    NaN        NaN      NaN
50%     472.000000   31.000000    NaN        NaN      NaN
75%     707.500000   43.000000    NaN        NaN      NaN
max     943.000000   73.000000    NaN        NaN      NaN

In [8]:
# Pregunta 8
df['occupation']

0         technician
1              other
2             writer
3         technician
4              other
           ...      
938          student
939    administrator
940          student
941        librarian
942          student
Name: occupation, Length: 943, dtype: object

In [9]:
# Pregunta 9
# Por el resumen dado en la pregunta 6 vemos que existen 21 tipos de ocupaciones contando la opción 'other'

# Pregunta 10
# Por el resumen dado en la pregunta 6 vemos que la ocupación más frecuente es 'student' con una frecuencia de 196.

# Pregunta 11
# Por el resumen dado en la pregunta 6 vemos que la edad media es de 31 años (mediana). 

# Pregunta 12

edad = df['age']
lista = list(edad.unique())
min = 1000
menosFrecuente = []
for i in range(len(lista)):
    c = list(edad).count(lista[i])
    if(c < min):
        min = c
        menosFrecuente = [lista[i]]
    elif(c == min):  
        menosFrecuente.append(lista[i])

print(min, menosFrecuente)

1 [7, 66, 11, 10, 73]


Las edades menos frecuentes son 7, 10, 11, 66 y 77, con una frecuencia igual a 1.